config/defaults.go

```diff
@@ -74,6 +74,9 @@ const (
 	// defaultFuseMetricsEmitWaitDurationSec is the amount of time the snapshotter will wait before emitting the metrics for FUSE operation.
 	defaultFuseMetricsEmitWaitDurationSec = 60
 
+	// defaultMaxConcurrency is the maximum number of layers allowed to be pulled at once
+	defaultMaxConcurrency = 100
```

Comment:
Chosen arbitrarily, lmk if this needs to be adjusted.

Reply:
I'm not sure how benchmarking would work in cases like these.

The preresolver has two tasks — deduplication and resource limiting. The former should be a strict performance increase and requires no tuning, but the resource limiting is what we are tuning with this variable. Finding a "best" value is a bit difficult here, since ultimately, we want the majority of customers to be unaware of this change, since they are not pulling too many layers at once. But, ideally, we also want to set a somewhat reasonable value for those who are doing such tasks.

I don't believe it's impossible, of course, I'm just unsure of how I'd go about finding a good value. Would love to hear more thoughts.

Reply:
can we run benchmark for a set of images for different values and check if the overall execution time is better or worse for different value?

Reply:
We actually want execution time to be slower whenever this value kicks in. We might instead want to see what memory usage looks like during a heavy image pull and fine-tune the value until it no longer crosses some threshold. I can discuss and see what that threshold might be.